e2e_tests/tests/experiment/test_custom_searcher.py

```diff
@@ -219,3 +226,305 @@ def sample_params(self) -> Dict[str, int]:
         hparams = {"global_batch_size": random.randint(10, 100)}
         logging.info(f"hparams={hparams}")
         return hparams
+
+
+@pytest.mark.e2e_cpu
+def test_run_asha_batches_exp() -> None:
+    config = conf.load_config(conf.fixtures_path("no_op/adaptive.yaml"))
+    config["searcher"] = {
+        "name": "custom",
+        "metric": "validation_error",
+        "smaller_is_better": True,
+        "unit": "batches",
+    }
+    config["name"] = "asha"
+    config["description"] = "custom searcher"
+
+    max_length = 3000
+    max_trials = 5
```

Comment:
with 3 rungs, we'd need at least 16 trials to see a trial on the top rung, so i suggest either 2 rungs with 8 trials or 3 rungs with 16 trials.  let's also see if we can get deterministic metrics so that we know exactly which and how many trials would be promoted to each rung and have checks to make sure things are as expected.

Reply:
we can get this behavior by configuring the no_op experiment config to have 0 std and a given metric_progression: https://github.com/determined-ai/determined/blob/custom-searcher/e2e_tests/tests/fixtures/no_op/model_def.py#L147

Reply:
Updated to 16 trials, 3 rungs, divisor 4. Added checks for number of batches processed by each trial to ensure promotion happens.